### PR TITLE
Sync v2-9-stable with v2-9-test to release 2.9.2

### PR DESCRIPTION
**Pull request by utkarsharma2**
Time for `2.9.2rc1`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,71 @@
...
+- Fix stacklevel for TaskContextLogger (#39142)
+
+Doc Only Changes
+""""""""""""""""
+- Provide extra tip on labeling DynamicTaskMapping (#39977)
+- Improve visibility of links / variables / other configs in Configuration Reference (#39916)
+- Remove 'legacy' definition for ``CronDataIntervalTimetable`` (#39780)
+- Update plugins.rst examples to use pyproject.toml over setup.py (#39665)
+- fix nit in pg set-up doc (#39628)
+- Docs: Update tracking-user-activity.rst (#39611)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Add Matomo to Tracking User Activity docs (#39611)
```

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,71 @@
...
+- Update best-practices.rst: fix Connection.get -> Connection. get_connection_from_secrets (#39560)
+- Adding note for provider dependencies (#39512)
+- Update docker-compose command (#39504)
+- Update note about restarting triggerer process (#39436)
+- Updating S3LogLink with an invalid bucket link (#39424)
+- Fix typo sensitive masking words in docs (#39415)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

Can skip this one on the changelog.

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: Resolved in here - https://github.com/apache/airflow/pull/40050/commits/c8bc544e0aa62662b9cc79c216f4e83ed5923036

**Review by jscheffl — APPROVED** [review]
Code from the 3 PRs/fixes from my side look good.

**Comment by Lee-W** [comment]
Looks like we'll need https://github.com/apache/airflow/commit/79042cff416296c8bdbd64f2e0a5f4d46c93e882 to pass the mypy check

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]
Looks good, LGTM

**Review by phanikumv — APPROVED** [review]

**Pull request merged**